src/main/java/com/qaprosoft/carina/demo/utils/AuthService.java

```diff
@@ -0,0 +1,60 @@
+package com.qaprosoft.carina.demo.utils;
+
+import com.qaprosoft.carina.core.foundation.webdriver.IDriverPool;
+import com.qaprosoft.carina.demo.gui.pages.AutomationPracticeCreateAccountPage;
+import com.qaprosoft.carina.demo.gui.pages.AutomationPracticeSignInPage;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.events.EventFiringWebDriver;
+
+import java.util.Random;
+
+public class AuthService implements IDriverPool {
+
+    private WebDriver getDriverSafe() {
```

Comment:
why you decided to implement here getDriverSafe()?

Reply:
I tried to use getDriver() and didnt work for me, so I looked up in a util class that is in the carina demo and saw that method, tried it and work. But it seem that I was doing something wrong, now I refactored to just getDriver() and works perfectly.